addons/web/static/src/core/utils/ui.js

```diff
@@ -227,3 +227,349 @@ export function addLoadingEffect(btnEl) {
+ * or a function that returns such an object.
+ * @param {number} [options.top] - The new top position of the element.
+ * @param {number} [options.left] - The new left position of the element.
+ * @param {Function} [options.using] - A function to execute after the position is set, 
+ * with the calculated position object passed as its argument.
+ * @param {number} [i] - Optional index of the element within a collection.
+ */
+function setOffset(elem, options, i) {
+    var curPosition, curLeft, curCSSTop, curTop, curOffset, curCSSLeft, calculatePosition,
+        position = getComputedStyle(elem).position,
+        curElem = elem.nodeType ? elem : elem[i],
+        props = {};
+
+    if (position === "static") {
+        elem.style.position = "relative";
+    }
+
+    curOffset = curElem.offset();
+    curCSSTop = getComputedStyle(elem).top;
+    curCSSLeft = getComputedStyle(elem).left;
+    calculatePosition = (position === "absolute" || position === "fixed") &&
+        (curCSSTop === "auto" || curCSSLeft === "auto");
+
+    if (calculatePosition) {
+        curPosition = position(curElem);
+        curTop = curPosition.top;
+        curLeft = curPosition.left;
+    } else {
+        curTop = parseFloat(curCSSTop) || 0;
+        curLeft = parseFloat(curCSSLeft) || 0;
+    }
+
+    if (typeof options === 'function') {
+        options = options(elem, i, Object.assign({}, curOffset));
+    }
+
+    if (options.top != null) {
+        props.top = (options.top - curOffset.top) + curTop;
+    }
+    if (options.left != null) {
+        props.left = (options.left - curOffset.left) + curLeft;
+    }
+
+    if (typeof options.using === 'function') {
+        options.using(elem, props);
+    } else {
+        Object.keys(props).forEach(function(key) {
+            elem.style[key] = props[key] + 'px';
+        });
+    }
+}
+
+/**
+ * @param {HTMLElement|string} el - the element to scroll to. If "el" is a
+ *      string, it must be a valid selector of an element in the DOM or
+ *      '#top' or '#bottom'. If it is an HTML element, it must be present
+ *      in the DOM.
+ *      Limitation: if the element is using a fixed position, this
+ *      function cannot work except if is the header (el is then either a
+ *      string set to '#top' or an HTML element with the "top" id) or the
+ *      footer (el is then a string set to '#bottom' or an HTML element
+ *      with the "bottom" id) for which exceptions have been made.
+ * @param {number} [options] - options for the scroll behavior
+ * @param {number} [options.extraOffset=0]
+ *      extra offset to add on top of the automatic one (the automatic one
+ *      being computed based on fixed header sizes)
+ * @param {number} [options.forcedOffset]
+ *      offset used instead of the automatic one (extraOffset will be
+ *      ignored too)
+ * @param {HTMLElement} [options.scrollable] the element to scroll
+ * @return {Promise}
+ */
+export function scrollTo(el, options = {}) {
```

Comment:
we have a utils/scrolling.js file (which already contains such a function)

Reply:
This scrollTo function is different from the one present in utils/scrolling.js file as it returns a promise that resolves when the scroll animation completes.